### DISCUSSION TITLE
writing-logs-to-elasticsearch-with-fluentd-using-serilog-in-asp-net-core/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#33] [Announcements]
# writing-logs-to-elasticsearch-with-fluentd-using-serilog-in-asp-net-core/

In this post, I describe how to customise the output of the Serilog Console sink so that you can pipe your console output to Elasticsearch using Fluentd.

https://andrewlock.net/writing-logs-to-elasticsearch-with-fluentd-using-serilog-in-asp-net-core/

**Comment by kyugoa** [comment]
testing this in openshift, the console sink was picking up by fluentd before adding the ElasticsearchJsonFormatter to the console output. the log message was parsed correctly and display as message field in Kibana. with the ElasticsearchJsonFormatter, the entire log message is serialized as JSON and now the entire JSON is displayed as message in Kibana. wonder if it's fluentd parsing it incorrectly.

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at June 21 2018, 04:46, in reply to [John Rutherford](https://disqus.com/by/disqus_GKiUL6X0mY/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/33#discussioncomment-6045992)</em>

---

<p>That wasn't my intention at all - using the Elasticsearch sink is a perfectly good option! It was just that in my case, Fluentd was already configured to pipe logs from the container's console output to Elasticsearch, so it made sense to reuse that capability, instead of using the sink directly. If that's not the case for you (quite likely), then the Elasticsearch sink is a perfectly good option!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at December 13 2019, 10:16, in reply to [Ignacio Fuentes](https://disqus.com/by/disqus_wNOdtQZcrC/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/33#discussioncomment-6046013)</em>

---

<p>In this example Fluentd is scraping the logs from the console, so there's no need for the fluentd sink</p>

[a comment by an automated account is not shown]

**Comment by behdad088** [comment]
Hi, How can I customize the fields? for example if I want the logs to be app.logs.{}, app.properties.{}, container-info.{}.
